src/payment/strategies/credit-card/credit-card-payment-strategy.spec.ts

```diff
@@ -196,4 +196,48 @@ describe('CreditCardPaymentStrategy', () => {
                 .toHaveBeenCalledWith(loadOrderAction);
         });
     });
+
+    describe('when hosted form is enabled but hosted fields are not present for rendering', () => {
+        let form: Pick<HostedForm, 'attach' | 'submit' | 'validate'>;
+        let initializeOptions: PaymentInitializeOptions;
+        let loadOrderAction: Observable<LoadOrderSucceededAction>;
+        let state: InternalCheckoutSelectors;
+
+        beforeEach(() => {
+            form = {
+                attach: jest.fn(() => Promise.resolve()),
+                submit: jest.fn(() => Promise.resolve()),
+                validate: jest.fn(() => Promise.resolve()),
+            };
+            initializeOptions = {
+                creditCard: {
+                    form: {
+                        fields: {},
+                    },
+                },
+                methodId: 'authorizenet',
+            };
+            loadOrderAction = of(createAction(OrderActionType.LoadOrderSucceeded, getOrder()));
+            state = store.getState();
+
+            jest.spyOn(state.config, 'getStoreConfig')
+                .mockReturnValue(merge(
+                    getConfig().storeConfig,
+                    { checkoutSettings: { isHostedPaymentFormEnabled: true } }
+                ));
+
+            jest.spyOn(orderActionCreator, 'loadCurrentOrder')
+                .mockReturnValue(loadOrderAction);
+
+            jest.spyOn(formFactory, 'create')
+                .mockReturnValue(form);
+        });
+
+        it('does not create hosted form', async () => {
```

Comment:
Can you also add a case to test that we're not submitting using the hosted form in this scenario?

Reply:
Done